rbc/irtools.py

```diff
@@ -136,9 +136,12 @@ def set_env(self, env_name, env):
 
 class JITRemoteTypingContext(typing.Context):
     def load_additional_registries(self):
-        from rbc.externals import math
+        from rbc.externals import cmath, libdevice
+        from rbc.omnisci_backend import mathimpl
```

Comment:
These imports have global effect (in the sense that they all exists in `sys.modules`), so I suggest moving these imports out of these method definitions.

In fact, we could do the registration automatically:
```python
import rbc.externals
    ...
    def load_additional_registries(self):
    for module in rbc.externals.values():
        if not isinstance(module, py_types.ModuleType):
            continue
        typing_registry = getattr(module, 'typing_registry', None)
        if typing_registry is not None:
            self.install_registry(typing_registry)
```

In this way, users can control which externals libraries to include by executing:
```
import rbc.externals.<modulename>
```
in their code. Although, this is going to be still have a global effect..

In sum, having the submodule imports in `rbc/externals/__init__.py` will allow introducing new external modules without editing irtools.py. Users can do that by:
```python
sys.modules['rbc.externals.mylib'] = <mylib module object>
```
(we may want to introduce `register_as_external(<module object>)`) as end-user interface but internally use the sys.modules trick.

Reply:
Done! I've created issue #298 to track `register_as_external` feature.